### PR TITLE
Refactor `ApproximateMostFrequent` to use type annotations

### LINKED ISSUE #21365: Use @TypeParameter to simplify approx_most_frequent aggregation function
With the merge of #21316 we can now simplify this aggregation function implementation by using @TypeParameter annotations in the serializer and state factory constructors, which means the function can be entirely converted to using the [function annotation framework](http://prestodb.io/docs/0.284/develop/functions.html). This will simplify the implementation and improve maintainability.

### PR DESCRIPTION
**Pull request by npochhi**
## Description
Resolves #21365. This PR refactors `ApproximateMostFrequent` function to use [function annotation framework](https://prestodb.io/docs/0.284/develop/functions.html). 

## Motivation and Context
This will simplify the implementation of the function.

## Impact
None

## Test Plan
Current tests

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java
Comment (ZacBlanco) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java:62]: nit: use static import for `BIGINT`

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java
Comment (ZacBlanco) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java:59]: nit: use static import for `BIGINT`

**Review by ZacBlanco — COMMENTED** [review]
two nits, otherwise LGTM

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java
Comment (npochhi) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java]: Done! Thanks for noticing this!

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java
Comment (npochhi) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/approxmostfrequent/ApproximateMostFrequent.java]: Done!

**Comment by ZacBlanco** [comment]
Can you please squash all changes into a single commit?

File: presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java
Comment (npochhi) [inline presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java:94]:
```suggestion
import com.facebook.presto.operator.aggregation.arrayagg.SetAggregationFunction;
import com.facebook.presto.operator.aggregation.arrayagg.SetUnionFunction;
```

**Comment by tdcmeehan** [comment]
LGTM, % need to rebase.  Please, once rebased, I can merge @npochhi.

File: presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java
Comment (npochhi) [reply presto-main/src/main/java/com/facebook/presto/metadata/BuiltInTypeAndFunctionNamespaceManager.java:875]: Yeah, sorry. Will resolve this first thing tomorrow.

**Comment by tdcmeehan** [comment]
@npochhi will you be rebasing?

**Comment by tdcmeehan** [comment]
FYI I just rebased to fix the merge conflict, will merge once tests are green.

**Review by tdcmeehan — APPROVED** [review]

**Comment by feilong-liu** [comment]
The aggregation function will fail for null input after this change, I am reverting the change in https://github.com/prestodb/presto/pull/21712, test case can be found in the reverting PR

**Comment by tdcmeehan** [comment]
Thanks @feilong-liu, reopened #21365.

**Pull request merged**
